src/nplinker/pairedomics/podp_antismash_downloader.py

```diff
@@ -25,81 +31,228 @@
 
 USER_AGENT = 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:86.0) Gecko/20100101 Firefox/86.0'
 
+GENOME_STATUS_FILENAME = "genome_status.txt"
 
-class GenomeStatus:
 
-    def __init__(self, original_id, resolved_id, attempted=False, filename=""):
+class GenomeStatus:
+    """Defines the status relative to a certain genome ID.
+
+    Attributes:
+        original_id(str): original ID for the genome. 
+        resolved_refseq_id(str): ID after having applied _resolve_refseq_id. 
+        attempted(bool): if the attempt of resolving the genome ID has already been done. Defaults to False. 
+        filename(str|PathLike): file in which the data for the genome, if present, are saved. Defaults to "". 
+        """
+
+    def __init__(self,
+                 original_id: str,
+                 resolved_refseq_id: str,
+                 attempted: str | bool = False,
+                 filename: str = ""):
         self.original_id = ';'.join(original_id.split(','))
-        self.resolved_id = None if resolved_id == 'None' else resolved_id
-        self.attempted = True if attempted == 'True' else False
+        self.resolved_refseq_id = "" if resolved_refseq_id == 'None' else resolved_refseq_id
+        if attempted == 'True':
+            self.attempted = True
+        else:
+            self.attempted = False
         self.filename = filename
 
-    @classmethod
-    def from_csv(cls, original_id, resolved_id, attempted, filename):
-        return cls(original_id, resolved_id, attempted, filename)
-
-    def to_csv(self):
-        return ','.join([
+    def add_line(self, file: Path):
+        """Adds a line to file, containing info for the current genome ID. 
+        """
+        line = ','.join([
             str(self.original_id),
-            str(self.resolved_id),
-            str(self.attempted), self.filename
+            str(self.resolved_refseq_id),
+            str(self.attempted),
+            str(self.filename)
         ])
 
+        with open(file, 'a+') as f:
+            f.write(line + '\n')
+
+
+def podp_download_and_extract_antismash_data(
+        genome_records: list[dict[str, dict[str, str] | str]],
+        project_download_cache: str | Path, project_file_cache: str | Path):
+
+    genome_status_file = Path(project_download_cache, GENOME_STATUS_FILENAME)
```

Comment:
Nice! Good to know that `Path` can accept mutiple paths and return the joined one.